### PR TITLE
Claire blades forms

### PR DESCRIPTION
**Pull request by clairecfu**
- [x] adding blades to RecursiveAssetSerializer
- [x] adding datacenter to RecursiveAssetSerializer and RecursiveAssetCPSerializer
- [x] adding conditional renderign to form to support chassis + chassis slots
- [x] loads available chassis slots 

todo:
- get blades for a chassis on change plan

### CODE REVIEW

File: src/forms/assetForm.tsx [hunk 1]
```diff
@@ -1095,357 +1106,216 @@ class AssetForm extends React.Component<AssetFormProps, AssetFormState> {
...
-                {values.model ? (
-                  <div>
-                    <Switch
-                      checked={this.state.customizeModel}
-                      label="Customize model values"
-                      onChange={this.handleCustomizeModelSwitch}
-                    />
+          {this.state.currDatacenter ? (
+            <div>
+              <Card>{this.renderModelFields()}</Card>
```
Comment (clairecfu) [inline src/forms/assetForm.tsx:1112]: did some refactoring in assetForm

File: rackcity/api/serializers/asset_serializers.py [hunk 2]
```diff
@@ -9,18 +9,22 @@
...
     NetworkPortCP,
 )
+from . import SiteSerializer
 from .it_model_serializers import ITModelSerializer
 from .rack_serializers import RackSerializer
 from .change_plan_serializers import GetChangePlanSerializer
 from rackcity.api.serializers.fields import RCIntegerField
 import copy
 
+from ...models.model_utils import ModelType
```
Comment (benauriemma) [inline rackcity/api/serializers/asset_serializers.py:19]: Can you change this to `rackcity.models.model_utils` pls - safer if stuff gets moved around

File: rackcity/api/serializers/asset_serializers.py [hunk 3]
```diff
@@ -342,6 +366,20 @@ def serialize_power_connections(power_port_model, asset):
...
+        return []
+
+    blades = Asset.objects.filter(chassis=asset.id)
+    serializer = AssetSerializer(blades, many=True,)
+    return serializer.data
+
+
+def get_blades_in_chassis_cp(asset_cp):
+    # TODO
+    return []
```
Comment (benauriemma) [inline rackcity/api/serializers/asset_serializers.py:380]: Assuming you have a task for this

File: rackcity/api/serializers/asset_serializers.py [hunk 4]
```diff
@@ -392,4 +430,12 @@ def get_neighbor_assets(hostname, id, nodes, edges, change_plan=None):
...
     except ObjectDoesNotExist:
-        return
\ No newline at end of file
+        return
+
+
+def get_datacenter_of_asset(asset):
+    if asset.rack:
+        datacenter = asset.rack.datacenter
+    if asset.chassis and asset.chassis.rack and asset.chassis.rack.datacenter:
```
Comment (benauriemma) [inline rackcity/api/serializers/asset_serializers.py:439]: I think `and asset.chassis.rack.datacenter` is unnecessary but it's also correct so I don't mind it

File: src/forms/assetForm.tsx [hunk 5]
```diff
@@ -286,7 +291,10 @@ class AssetForm extends React.Component<AssetFormProps, AssetFormState> {
     } = asset;
     const model = asset.model ? asset.model.id : null;
     const rack = asset.rack ? asset.rack.id : null;
+    const chassis = asset.chassis? asset.chassis.id:null;
```
Comment (benauriemma) [inline src/forms/assetForm.tsx:294]: Ben linter: 2
Claire linter: 1

File: src/forms/assetForm.tsx [hunk 6]
```diff
@@ -1497,6 +1367,254 @@ class AssetForm extends React.Component<AssetFormProps, AssetFormState> {
...
+                      values.chassis_slot? values.chassis_slot
+                        : "Chassis Slot"
+                    }
+                  />
+                </StringSelect>
+      </FormGroup>
+    ];
+  }
+  getChassisSlots(){
+    let slots = ["1","2","3","4","5","6","7","8","9","10","11","12"]
```
Comment (benauriemma) [inline src/forms/assetForm.tsx:1610]: You've peaked

File: rackcity/api/serializers/asset_serializers.py (hunk 2 above)
Comment (clairecfu) [reply rackcity/api/serializers/asset_serializers.py:19]: pycharm did this 😠

File: rackcity/api/serializers/asset_serializers.py (hunk 3 above)
Comment (clairecfu) [reply rackcity/api/serializers/asset_serializers.py:380]: yeet mental task but i promise i will make a notion task

File: rackcity/api/serializers/asset_serializers.py (hunk 2 above)
Comment (benauriemma) [reply rackcity/api/serializers/asset_serializers.py]: Gross!

File: rackcity/api/serializers/asset_serializers.py (hunk 3 above)
Comment (benauriemma) [reply rackcity/api/serializers/asset_serializers.py:380]: Pinky promise?

Comment (clairecfu) [reply rackcity/api/serializers/asset_serializers.py:380]: image

Comment (clairecfu) [reply rackcity/api/serializers/asset_serializers.py:380]: :)

File: src/forms/assetForm.tsx (hunk 6 above)
Comment (clairecfu) [reply src/forms/assetForm.tsx:1610]: thank u

File: rackcity/api/serializers/asset_serializers.py (hunk 3 above)
Comment (benauriemma) [reply rackcity/api/serializers/asset_serializers.py:380]: 🥰

**Pull request merged**